app/services/mongodb/MongoDBQueueService.scala

```diff
@@ -68,38 +69,36 @@ trait MongoDBQueueService {
     }
   }
 
-  // get next entry from queue
-  def getNextQueuedAction(): Option[QueuedAction] = {
+  // get next entries from queue
+  def getBatchQueuedAction(): List[QueuedAction] = {
     try {
-      val response = Queue.findOne(new MongoDBObject)
+      val response = Queue.find(new MongoDBObject).limit(batchSize)
       if (queueFetchError) {
         Logger.info("MongoDB has successfully reconnected.")
         queueFetchError = false
       }
-      response
+      response.toList
     } catch {
       case e: MongoException => {
         // Only log an error once on failed fetch, instead of repeating every 5ms
         if (!queueFetchError) {
           Logger.error("Problem connecting to MongoDB queue.")
           queueFetchError = true
         }
-        None
+        List.empty[QueuedAction]
       }
-      case _ => None
+      case _ => List.empty[QueuedAction]
     }
-
   }
 
-  // start pool to being processing queue actions
+  // start processing queue actions
   def listen() = {
-    if (queueTimer == null) {
-      // TODO: Need to make these in a separate pool
-      queueTimer = Akka.system().scheduler.schedule(0 seconds, 5 millis) {
-        getNextQueuedAction match {
-          case Some(qa) => handleQueuedAction(qa)
-          case None => {}
-        }
+    if (ec == null) {
+      ec = ExecutionContext.fromExecutor(new java.util.concurrent.ForkJoinPool(threads))
```

Comment:
What's the difference between this and creating and Akka context with `implicit val myExecutionContext: ExecutionContext = Akka.system.dispatchers.lookup("my-context")
`?

I looked it up a bit but couldn't find a clear answer. Thank you.

Reply:
This is also an interesting read https://jessitron.com/2014/01/29/choosing-an-executorservice/.

Reply:
only issue I saw/had was when using the loopup you need to define it in your app.config, otherwise it throws an error. in this case it creates one for N threads, and runs the indexer in parallel. This was hoping to be able to process a large number of waiting messages quickly using multiple threads.